src/lib.rs

```diff
@@ -263,7 +322,7 @@ impl<F> Server<F> where F: Send + Sync + 'static + Fn(&Request) -> Response {
     /// Runs the server forever, or until the listening socket is somehow force-closed by the
     /// operating system.
     #[inline]
-    pub fn run(self) {
+    pub fn run(&self) {
```

Comment:
What is the reason of this change?

Reply:
This is so a call to `run` can be chained after a `pool_size`:

```rust
Server::new("localhost:8000", handler)?
    .pool_size(8 * num_cpus::get())
    .run()
```

Reply:
I'd prefer that `pool_size` takes by `self` and returns a `Self`.

Reply:
Alright, `pool_size` and `run` signatures have been changed to take/return `self`.